beetsplug/yamleditor.py

```diff
@@ -0,0 +1,329 @@
+            'albumfields': 'album albumartist',
+            'itemfields': 'track title artist album ',
+            'not_fields': 'path',
+            'separator': '-'
+
+        })
+
+    def commands(self):
+        yamleditor_command = Subcommand(
+            'yamleditor',
+            help='send items to yamleditor for editing tags')
+        yamleditor_command.parser.add_option(
+            '-e', '--extra',
+            action='store',
+            help='add additional fields to edit',
+        )
+        yamleditor_command.parser.add_all_common_options()
+        yamleditor_command.func = self.editor_music
+        return[yamleditor_command]
+
+    def editor_music(self, lib, opts, args):
+        """edit tags in a textfile in yaml-style
+        """
+        self.style = self.config['style'].get()
+        """the editor field in the config lets you specify your editor.
+        Defaults to open with webrowser module"""
+        self.editor = self.config['editor'].get()
+        """the editor_args field in your config lets you specify
+        additional args for your editor"""
+        self.editor_args = self.config['editor_args'].get().split()
+        """the html_viewer field in your config lets you specify
+        your htmlviewer. Defaults to open with webrowser module"""
+        self.html_viewer = self.config['html_viewer'].get()
+        """the html_args field in your config lets you specify
+        additional args for your viewer"""
+        self.html_args = self.config['html_args'].get().split()
+        """the diff_method field in your config picks the way to see your
+         changes. Options are:
+        'ndiff'(2 files with differences),
+        'unified'(just the different lines and a few lines of context),
+        'html'(view in html-format),
+        'vimdiff'(view in VIM)"""
+        self.diff_method = self.config['diff_method'].get()
+        """the albumfields field in your config sets the tags that
+        you want to see/change for albums.
+        Defaults to album albumartist.
+        the ID tag will always be listed as it is used to identify the item"""
+        self.albumfields = self.config['albumfields'].get().split()
+        """the itemfields field in your config sets the tags that
+        you want to see/change or items.
+        Defaults to track title artist album.
+        the ID tag will always be listed as it is used to identify the item"""
+        self.itemfields = self.config['itemfields'].get().split()
+        '''the not_fields field in your config sets the tags that
+        will not be changed.
+        If you happen to change them, they will be restored to the original
+        value. The ID of an item will never be changed.'''
+        self.not_fields = self.config['not_fields'].get().split()
+        '''the separator in your config sets the separator that will be used
+        between fields in your terminal. Defaults to -'''
+        self.separator = self.config['separator'].get()
```

Comment:
These configuration lookups should move either to the `__init__` method (so they run when the plugin is loaded) or get distributed throughout the code to be close to where they're used. Assigning to fields on `self` can cause some unexpected conflicts.